src/ngraph/runtime/cpu/pass/cpu_rnn_fusion.cpp

```diff
@@ -160,13 +154,17 @@ void ngraph::runtime::cpu::pass::LSTMFusion::construct_onnx_lstmcell_fprop()
             return std::move(new_bias);
         };
 
-        auto W_iofc = pattern_map[W];
-        auto R_iofc = pattern_map[R];
-        auto W_ifco = get_weights_ifco_gate_order(W_iofc);
-        auto R_ifco = get_weights_ifco_gate_order(R_iofc);
-        // here onnx bias will be of shape (2 * gates_count * hidden_size) bias of Wb and Rb are
-        // concatenated, we will split the bias, add and rearrange in order IFCO
-        auto bias_ifco = get_bias_ifco_gate_order(bias_iofc);
+        // we need to reorder W, R and bias from IOFC to IFCO gate order
+        // Note: ONNX runtime provides W, R and bias in the gate order [IOFC] but
+        // MKLDNN computes LSTM kernel in the [IFCO] order.
+        if (lstmcell_op->get_weights_format() != op::LSTMWeightsFormat::IFCO)
```

Comment:
@pruthviIntel Only the following test cases should run with MKLDNN on CPU backend:
* [lstm_cell_no_bias_no_peepholes](https://github.com/NervanaSystems/ngraph/blob/tsocha/fused-lstm/test/backend/fused_op.in.cpp#L1286-L1346)
* [lstm_cell_zero_bias_peepholes_constant](https://github.com/NervanaSystems/ngraph/blob/tsocha/fused-lstm/test/backend/fused_op.in.cpp#L1419-L1484),
since they're passing last two inputs as `Constant` which satisfies [this condition](https://github.com/NervanaSystems/ngraph/blob/tsocha/fused-lstm/src/ngraph/runtime/cpu/cpu_external_function.cpp#L1199). 
When CPU backend has support for `LSTMCell` then the op decomposition is **not** run. I added the check you cited to make sure we're not converting weights which are already in correct order.